core/web/job_specs_controller.go

```diff
@@ -49,25 +49,45 @@ func (jsc *JobSpecsController) requireImplented(js models.JobSpec) error {
 	return nil
 }
 
+// getAndCheckJobSpec(c) returns a validated job spec from c, or errors. On
+// error, a non-zero HTTP status is returned.
+func (jsc *JobSpecsController) getAndCheckJobSpec(
+	c *gin.Context) (js models.JobSpec, httpStatus int, err error) {
+	var jsr models.JobSpecRequest
+	if err := c.ShouldBindJSON(&jsr); err != nil {
+		// TODO(alx): Better parsing and more specific error messages
+		// https://www.pivotaltracker.com/story/show/171164115
+		return models.JobSpec{}, http.StatusBadRequest, err
+	}
+	js = models.NewJobFromRequest(jsr)
+	if err := jsc.requireImplented(js); err != nil {
+		return models.JobSpec{}, http.StatusNotImplemented, err
+	}
+	if err := services.ValidateJob(js, jsc.App.GetStore()); err != nil {
+		return models.JobSpec{}, http.StatusBadRequest, err
+	}
+	return js, 0, nil
+}
+
 // Create adds validates, saves, and starts a new JobSpec.
 // Example:
 //  "<application>/specs"
 func (jsc *JobSpecsController) Create(c *gin.Context) {
-	var jsr models.JobSpecRequest
-	if err := c.ShouldBindJSON(&jsr); err != nil {
-		jsonAPIError(c, http.StatusBadRequest, err)
-	} else if js := models.NewJobFromRequest(jsr); false {
-	} else if err := jsc.requireImplented(js); err != nil {
-		jsonAPIError(c, http.StatusNotImplemented, err)
-	} else if err := services.ValidateJob(js, jsc.App.GetStore()); err != nil {
-		jsonAPIError(c, http.StatusBadRequest, err)
-	} else if err := NotifyExternalInitiator(js, jsc.App.GetStore()); err != nil {
+	js, httpStatus, err := jsc.getAndCheckJobSpec(c)
+	if err != nil {
+		jsonAPIError(c, httpStatus, err)
+		return
+	}
+	if err := NotifyExternalInitiator(js, jsc.App.GetStore()); err != nil {
 		jsonAPIError(c, http.StatusInternalServerError, err)
-	} else if err = jsc.App.AddJob(js); err != nil {
+		return
+	}
+	if err := jsc.App.AddJob(js); err != nil {
 		jsonAPIError(c, http.StatusInternalServerError, err)
-	} else {
-		jsonAPIResponse(c, presenters.JobSpec{JobSpec: js}, "job")
+		return
 	}
+	// https://www.pivotaltracker.com/story/show/171169052
+	jsonAPIResponse(c, presenters.JobSpec{JobSpec: js}, "job")
```

Comment:
Why is the tracker story left in the codebase?  If its necessary to leave in, commenting it would answer the same question to everyone else reading the code.

Reply:
That's true, but there's a concern that the comment will go stale, or the next person will feel obliged to fulfill the TODO. @se3000 proposed adding `TODO:` to the comment, so I've done that.